yaksh/views.py

```diff
@@ -608,7 +610,10 @@ def course_detail(request, course_id):
     ci = RequestContext(request)
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = Course.objects.filter(Q(creator=user)|Q(teachers=user),
```

Comment:
If in case course is None then next statement course.creator will throw an error.